internal/film/filmrepo_els.go

```diff
@@ -0,0 +1,61 @@
+package film
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/PhamDuyKhang/littledetective/internal/types"
+	"github.com/olivere/elastic"
+	"reflect"
+)
+
+type (
+	ElasticFilmRepository struct {
+		client *elastic.Client
+	}
+)
+
+func NewElasticFilmRepository(c *elastic.Client) *ElasticFilmRepository {
+	return &ElasticFilmRepository{
+		client: c,
+	}
+}
+func (el *ElasticFilmRepository) Search(text string) (types.SearchResult, error) {
+	var resultType types.Film
+	termQuery := elastic.NewQueryStringQuery(text)
+	searchResult, err := el.client.Search().
+		Index("imdb").
+		Query(termQuery).
+		Pretty(true).
+		Do(context.Background())
```

Comment:
Should read about package `context` and the reason why Elasticsearch API support context here.

Reply:
I will read it about reason why Elasticsearch API support context